src/robot/variables/scopes.py

```diff
@@ -211,10 +211,14 @@ def _convert_cli_variable(self, name, typ, value):
             info = TypeInfo.from_variable(var)
         except DataError as err:
             raise DataError(f"Invalid command line variable '{var}': {err}")
+        from robot.api import Secret
```

Comment:
Imports that cannot be at the top of the module should be at the top of the function/method. In this case there's already another import there as well.